plugins/resolver.go

```diff
@@ -0,0 +1,63 @@
+package plugins
+
+import (
+	"context"
+	"fmt"
+	"sort"
+
+	"github.com/els0r/goProbe/v4/pkg/distributed/hosts"
+)
+
+// ResolverInitializer constructs a resolver, optionally using a config file.
+// Mirrors the existing QuerierInitializer pattern
+type ResolverInitializer func(ctx context.Context, cfgPath string) (hosts.Resolver, error)
+
+// RegisterResolver is typically called from init() in a resolver package.
+func RegisterResolver(name string, initFn ResolverInitializer) {
+	GetInitializer().registerResolver(name, initFn)
+}
+
+// GetAvailableResolverPlugins returns a list of all registered resolver plugins
+func GetAvailableResolverPlugins() []string {
+	return GetInitializer().getResolvers()
+}
+
+func (i *Initializer) getResolvers() []string {
+	plugins := make([]string, 0)
```

Comment:
Supa-nit: Could put inside `RLock()` and pre-initialize with `len(i.resolvers)`.